<launch>
	<!-- fake a primesense point cloud from a pcd file for testing placed somewhere in the world -->	
	<remap from="cloud_pcd" to="/camera/depth_registered/points"/>
	<node pkg="pcl_ros" name="fake_primesense" type="pcd_to_pointcloud" respawn="true" output="screen" args="$(find plane_segmentation)/resources/scene.pcd 0.1 _frame_id:=/camera_link"/>
	<node pkg="tf" type="static_transform_publisher" name="kinect_base_link5" args="0 0 0 0 0 0 1  /camera_link /world 100"/>

	<!-- the plane detector -->
	<node pkg="plane_segmentation" name="curvature_filter" type="curvature_filter_node" respawn="true" output="screen">
		<!--topic remapping-->
		<!-- <remap from="input_cloud" to="/camera/depth_registered/points" /> -->
		<param name="voxel_size" value="0.02" />
		<param name="normal_radius" value="0.05" />
		<param name="curvature_threshold" value="0.01" />
		<param name="min_cluster_size" value="30" />
	</node>

	<!-- load a customized rviz -->
	<node name="$(anon rviz)" launch-prefix="" pkg="rviz" type="rviz" respawn="false" args="-d $(find plane_segmentation)/launch/rviz.rviz" output="screen"/>

</launch>
